### ISSUE TITLE
Unclear error messages in command line output

### ISSUE THREAD

**Issue by orlade** [anz-bank/sysl#5]
When sysl fails, it typically does so in a meaningless way. For example:

```
$ sysl data --output ../sysl-out/petshop.data.txt //demo/petshop/petshop
Traceback (most recent call last):
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysl.py", line 136, in <module>
    main()
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysl.py", line 131, in main
    args.func(args)
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysldata.py", line 168, in cmd
    out = dataviews(module, args)
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysldata.py", line 125, in dataviews
    parts = re.match(ur'^(.*?)(?:\s*<-\s*(.*?))?$', args.project)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer
```

This appears to perhaps be complaining about a missing `--project` arg, but it's not at all clear. Adding `--project petshop` (at a guess) produces this:

```
$ sysl data --output ../sysl-out/petshop.data.txt --project petshop //demo/petshop/petshop
Traceback (most recent call last):
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysl.py", line 136, in <module>
    main()
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysl.py", line 131, in main
    args.func(args)
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysldata.py", line 168, in cmd
    out = dataviews(module, args)
  File "/Users/Oliver/dev/sysl/bazel-bin/src/sysl/sysl.runfiles/com_github_anz_bank_sysl/src/sysl/sysldata.py", line 132, in dataviews
    endpts = app.endpoints.itervalues()
AttributeError: 'NoneType' object has no attribute 'endpoints'
```

This appears to be complaining that the `petshop` project doesn't exist, but again not clear if that's the case, and if so how to fix it.

First of all, the docs should make it clear what arguments are required, and what they should refer to. However in the event that something critical is missing, the error should be human-readable.

**Comment by marcelocantos** [comment]
It is technically meaningful. However, it's far from obvious to the casual user what's gone wrong.

Last week, we cleaned up the latest internal code base in order to do another push to the open source repo in the next day or two. In the same breath, we will flip the switch so that the open source code base becomes the point of truth. Once this is done, cleaning up the user experience will become the major target of our efforts.

**Comment by juliaogris** [comment]
@orlade there is now PR #26 addressing your concerns:
```
$ sysl data --output ../sysl-out/petshop.data.txt //demo/petshop/petshop
usage: sysl data [-h] --project PROJECT [--title TITLE] [--plantuml PLANTUML]
                 [--verbose] [--expire-cache] [--dry-run] [--filter FILTER]
                 --output OUTPUT
                 modules [modules ...]
sysl data: error: argument --project/-j is required
```
and
```
$ sysl data --output ../sysl-out/petshop.data.txt --project petshop //demo/petshop/petshop
Traceback (most recent call last):
  File "/Users/ogrisj/.virtualenvs/github-sysl/bin/sysl", line 11, in <module>
    load_entry_point('sysl', 'console_scripts', 'sysl')()
  File "/Users/ogrisj/dev/github-sysl/src/sysl/core/__main__.py", line 121, in main
    args.func(args)
  File "/Users/ogrisj/dev/github-sysl/src/sysl/core/sysldata.py", line 163, in cmd
    out = dataviews(module, args)
  File "/Users/ogrisj/dev/github-sysl/src/sysl/core/sysldata.py", line 119, in dataviews
    raise Exception('Invalid project "{}"'.format(args.project))
Exception: Invalid project "petshop"
```
Admittedly, the last error message is not ideal but an improvement I believe.
Further help can be found with `sysl data --help` which lists amongst others:
```
 --project PROJECT, -j PROJECT
                        project pseudo-app to render
                        eg: PetShopModel, "ATM <- GetBalance"
```
I've also added some updates to the [README](https://github.com/juliaogris/sysl/blob/error-msg/README.rst#installation).
I will try to work towards a more digestible Exception representation in a follow-up PR which should address the `[... ugly stack trace ...] Exception: Invalid project "petshop"`.
